src/Compilers/Core/Portable/CodeAnalysisResources.resx

```diff
@@ -519,4 +519,7 @@
   <data name="TuplesNeedAtLeastTwoElements" xml:space="preserve">
     <value>Tuples must have at least two elements.</value>
   </data>
+  <data name="TupleUnderlyingTypeMustBeTupleCompatible" xml:space="preserve">
+    <value>The underlying type for a tuple must have the shape of System.ValueTuple.</value>
+  </data>
```

Comment:
Aleksey, I may need some help to more clearly explain what is "tuple compatible".